frontend/components/ExportBanner.js

```diff
@@ -29,20 +30,37 @@ const Square = ({ fill }) => html`
 //@ts-ignore
 window.enable_secret_pluto_recording = true
 
-export const ExportBanner = ({ onClose, notebookfile_url, notebookexport_url, start_recording }) => {
+export const ExportBanner = ({ notebook_id, onClose, notebookfile_url, notebookexport_url, start_recording }) => {
+    const isDesktop = !!window.plutoDesktop
+
+    const exportNotebook = (/** @type {{ preventDefault: () => void; }} */ e, /** @type {0 | 1 | 2 | 3} @note see *types.d.ts* for more info.*/ type) => {
+        if (isDesktop) {
+            e.preventDefault()
+            window.plutoDesktop?.fileSystem.exportNotebook(notebook_id, type)
+        }
+    }
+
+    const elType = isDesktop ? "button" : "a"
+
     return html`
         <aside id="export">
             <div id="container">
                 <div class="export_title">export</div>
                 <!-- no "download" attribute here: we want the jl contents to be shown in a new tab -->
-                <a href=${notebookfile_url} target="_blank" class="export_card">
+                <${elType} href=${notebookfile_url} target="_blank" class="export_card" onclick=${(e) => exportNotebook(e, 0)}>
                     <header><${Triangle} fill="#a270ba" /> Notebook file</header>
                     <section>Download a copy of the <b>.jl</b> script.</section>
-                </a>
-                <a href=${notebookexport_url} target="_blank" class="export_card" download="">
+                </${elType}>
+                <${elType}
```

Comment:
I'm changing the nice `eltype` = `isDesktop ? "button" : "a"` back to just an `a` tag, based on my small test that works just fine. Perhaps the problem was using `onclick` vs `onClick` in the attribute.

Reply:
got it!